tuya.cpp

```diff
@@ -228,31 +227,81 @@ void DeRestPluginPrivate::handleTuyaClusterIndication(const deCONZ::ApsDataIndic
             quint8 hour;
             quint8 minut;
             quint8 heatSetpoint;
+            
+            quint16 minut16;
+            quint16 heatSetpoint16;
 
-            quint8 part = 1;
+            quint8 part = 0;
             QList<int> listday;
-
-            if (dp == 0x0070) //work days (6)
-            {
-                listday << 124;
-            }
-            else if (dp == 0x00071) // holiday = Not working day (6)
-            {
-                listday << 003;
-            }
-            else if (dp == 0x0065) // Moe thermostat W124 (4) + W002 (4) + W001 (4)
+            
+            switch (dp)
             {
-                part = length / 3;
-                listday << 124 << 2 << 1;
-            }
+                case 0x0070: //work days (6)
+                {
+                    part = 1;
+                    listday << 124;
+                    length = length / 3;
+                }
+                break;
+                case 0x0071: // holiday = Not working day (6)
+                {
+                    part = 1;
+                    listday << 003;
+                    length = length / 3;
+                }
+                break;
+                case 0x0065: // Moe thermostat W124 (4) + W002 (4) + W001 (4)
+                {
+                    part = length / 3;
+                    listday << 124 << 2 << 1;
+                    length = length / 3;
+                }
+                break;
+                // Daily schedule (mode 8)(minut 16)(temperature 16)(minut 16)(temperature 16)(minut 16)(temperature 16)(minut 16)(temperature 16)
+                case 0x007B: // Sunday
+                case 0x007C: // Monday
+                case 0x007D: // Thuesday
+                case 0x007E: // Wednesday
+                case 0x007F: // Thursday
+                case 0x0080: // Friday
+                case 0x0081: // Saturday
+                {
+                    int t[8] = {1,64,32,46,8,4,2};
+                    part = 1;
+                    listday << t[dp-123];
```

Comment:
This is dangerous in terms of maintenance if the code is extended later, the `dp - 123` index isn't bound checked.
Albeit due the switch case values a bound check is there, I'd prefer a more specific version. It took me a few minutes to realize that there is already an implicit array bound check and it's not a SEGV canditate.

```
const std::array<int, 7> t = {1,64,32,46,8,4,2};

if (dp < 0x007B || (dp - 0x007B) >= t.size())
{
    DPG_Printf(DBG_INFO, "Tuya unsupported daily schedule dp value: 0x%04X\n", dp);
    return; // bail out early
}
```

Note I replaced 123 by the equal hex value 0x007B to make it more clear where this lower bound offset (Sunday) comes from.